### PR TITLE
add rrddim_get_last_stored_value to simplify function code in internal collectors

### PR DESCRIPTION
**Pull request by ilyam8**
##### Summary

ssia

##### Test Plan

Check internal collectors functions.

##### Additional Information

<details> <summary>For users: How does this change affect me?</summary>
  
</details>

### CODE REVIEW

**Review by thiagoftsm — DISMISSED** [review]
PR is working as expected, LGTM!

File: collectors/diskspace.plugin/plugin_diskspace.c [hunk 1]
```diff
@@ -685,12 +669,12 @@ int diskspace_function_mount_points(BUFFER *wb, int timeout __maybe_unused, cons
...
-        double space_avail = get_last_stored_value(mp->rd_space_avail, &max_space_avail, 1.0, 1.0);
-        double space_used = get_last_stored_value(mp->rd_space_used, &max_space_used, 1.0, 1.0);
-        double space_reserved = get_last_stored_value(mp->rd_space_reserved, &max_space_reserved, 1.0, 1.0);
-        double inodes_avail = get_last_stored_value(mp->rd_space_avail, &max_space_avail, 1.0, 1.0);
-        double inodes_used = get_last_stored_value(mp->rd_space_used, &max_space_used, 1.0, 1.0);
-        double inodes_reserved = get_last_stored_value(mp->rd_space_reserved, &max_space_reserved, 1.0, 1.0);
+        double space_avail = rrddim_get_last_stored_value(mp->rd_space_avail, &max_space_avail, 1.0);
+        double space_used = rrddim_get_last_stored_value(mp->rd_space_used, &max_space_used, 1.0);
+        double space_reserved = rrddim_get_last_stored_value(mp->rd_space_reserved, &max_space_reserved, 1.0);
+        double inodes_avail = rrddim_get_last_stored_value(mp->rd_space_avail, &max_space_avail, 1.0);
```
Comment (stelfrag) [inline collectors/diskspace.plugin/plugin_diskspace.c:675]: Please check the inodes calculations

**Review by vkalintiris — CHANGES_REQUESTED** [review]
@ilyam8 I'm pedantic with this one because it touches the API of `database/`.

File: database/rrd.h [hunk 2]
```diff
@@ -1668,6 +1668,22 @@ void rrdset_pluginsd_receive_unslot_and_cleanup(RRDSET *st);
 void rrdset_pluginsd_receive_unslot(RRDSET *st);
 
 // ----------------------------------------------------------------------------
+static inline double rrddim_get_last_stored_value(RRDDIM *rd_dim, double *max_value, double div) {
```
[the remark this answers is not in the document]
Comment (ilyam8) [reply database/rrd.h:1671]: > or even better create a new file collectors/common.h

I agree with that. I tried to make it static and put in `collectors/all.h` but it dodn't work. I asked on Slack and was told to put in rrd.h which I did. @stelfrag @MrZammler if you can address @vkalintiris request please help and push a commit to this PR.

**Review by vkalintiris — APPROVED** [review]
LGTM considering that doing the changes I  asked is not trivial due to the way we currently organize headers. I'll try to cleanup the headers myself in the future.

**Review by stelfrag — APPROVED** [review]

**Pull request merged**